Wrapping/Generators/Python/itk/support/helpers.py

```diff
@@ -173,3 +176,127 @@ def image_filter_wrapper(*args, **kwargs):
+        if os.name == 'nt':
+            return 'uint32_t', 1
+        else:
+            return 'uint64_t', 1
+    mangle = None
+    pixelType = 1
+    if component == itk.SL:
+        if os.name == 'nt':
+            return 'int32_t', 1,
+        else:
+            return 'int64_t', 1,
+    if component in (itk.SC, itk.UC, itk.SS, itk.US, itk.SI, itk.UI, itk.F,
+            itk.D, itk.B, itk.SL, itk.SLL, itk.UL, itk.ULL):
+        mangle = component
+    elif component in [i[1] for i in itk.Vector.items()]:
+        mangle = itk.template(component)[1][0]
+        pixelType = 5
+    elif component == itk.complex[itk.F]:
+        # complex float
+        return 'float', 10
+    elif component == itk.complex[itk.D]:
+        # complex float
+        return 'double', 10
+    elif component in [i[1] for i in itk.CovariantVector.items()]:
+        # CovariantVector
+        mangle = itk.template(component)[1][0]
+        pixelType = 7
+    elif component in [i[1] for i in itk.Offset.items()]:
+        # Offset
+        return 'int64_t', 4
+    elif component in [i[1] for i in itk.FixedArray.items()]:
+        # FixedArray
+        mangle = itk.template(component)[1][0]
+        pixelType = 11
+    elif component in [i[1] for i in itk.RGBAPixel.items()]:
+        # RGBA
+        mangle = itk.template(component)[1][0]
+        pixelType = 3
+    elif component in [i[1] for i in itk.RGBPixel.items()]:
+        # RGB
+        mangle = itk.template(component)[1][0]
+        pixelType = 2
+    elif component in [i[1] for i in itk.SymmetricSecondRankTensor.items()]:
+        # SymmetricSecondRankTensor
+        mangle = itk.template(component)[1][0]
+        pixelType = 8
+    else:
+        raise RuntimeError('Unrecognized component type: {0}'.format(str(component)))
+    _python_to_js = {
+        itk.SC: 'int8_t',
+        itk.UC: 'uint8_t',
+        itk.SS: 'int16_t',
+        itk.US: 'uint16_t',
+        itk.SI: 'int32_t',
+        itk.UI: 'uint32_t',
+        itk.F: 'float',
+        itk.D: 'double',
+        itk.B: 'uint8_t',
+        itk.SL: 'int64_t',
+        itk.UL: 'uint64_t',
+        itk.SLL: 'int64_t',
+        itk.ULL: 'uint64_t',
+    }
+    imageType = dict(
+        dimension=itkimage.GetImageDimension(),
+        componentType=_python_to_js[mangle],
+        pixelType=pixelType,
+        components=itkimage.GetNumberOfComponentsPerPixel()
+    )
+    return imageType
+
+
+def image_type_from_wasm_type(jstype):
+    import itk
+
+    _pixelType_to_prefix = {
+        1: '',
+        2: 'RGB',
+        3: 'RGBA',
+        4: 'O',
+        5: 'V',
+        7: 'CV',
+        8: 'SSRT',
+        11: 'FA'
+    }
+    pixelType = jstype['pixelType']
+    dimension = jstype['dimension']
+    if pixelType == 10:
+        if jstype['componentType'] == 'float':
+            return itk.Image[itk.complex, itk.F], np.float32
+        else:
+            return itk.Image[itk.complex, itk.D], np.float64
+
+    def _long_type():
```

Comment:
@GenevieveBuckley you could use this or something similar to treat `long` different depending on platform.

Reply:
Thank you, I had seen this. My question was more about what result should be returned for different platforms, eg:

I don't have access to a Windows machine, so I had to guess these:
* nt, signed long  -> ?? (I guessed `int64_t`, but from the discussion about it sounds like it should be `int32_t`)
* nt, signed long long -> ?? (I guessed `int64_t`, and I don't see anyone disagreeing)
* nt, unsigned long -> ?? (I guessed `uint64_t`, but from the discussion about it sounds like it should be `uint32_t`)
* nt, unsigned long long -> ?? (I guessed `uint64_t`, and I don't see anyone disagreeing)

I know the right answers on posix, because I can check these on my machine.
* posix, signed long -> should be `int64_t`
* posix, signed long long -> should be `int64_t`
* posix, unsigned long -> should be `uint64_t`
* posix, unsigned long long -> should be `uint64_t`